is_equal/is_equal.tex

```diff
@@ -0,0 +1,155 @@
+    output_domain = (VectorDomain(AllDomain(bool));
+    input_metric = metric; 
+    let output_metric = metric
+    assert is_instance(metric, SymmetricDistance);
+    
+    let function(data: Vec<T>) -> Vec<Bool>: 
+        return map(data == value);
+    let stability_relation = lambda(d_in, d_out): (d_in <= d_out);
+    
+    return Transformation(input_domain, output_domain, function, input_metric, output_metric, stability_relation)
+
+Is_Equal = make_is_equal(value, SymmetricDistance);
+\end{lstlisting}
+
+
+\subsection{Conditions as Specified in Pseudo Code}
+\begin{itemize}
+    \item Input Domain: all vector domain of input value type \texttt{T} (same type as input \texttt{value}). T is a generic type that is static and can be evaluated with equality.
+    \item Output Domain: all vector domain of type \texttt{bool}.
+    \item Function: returns vector mapping of whether vector elements equal input \texttt{value}.
+    \item Input Metric: Symmetric Distance
+    \item Output Metric: same as Input Metric. 
+    \item Stability Relation: function that takes in 32 bit integers $d_{in}$ and $d_{out}$ and returns whether $d_{in} \leq d_{out}.$
+\end{itemize}
+
+\silvia{Do we need to include the stability parameter? And perhaps include an example calling \texttt{make\_is\_equal} with metrics?}
+
+\silvia{Shouldn't it be checked / imposed that the end user passes the same input and out metric somewhere? Otherwise have only one?}
+
+\subsection{Proof}
+
+\begin{theorem}
+For every setting of the input parameters to \texttt{make\_is\_equal}, the transformation returned by \texttt{make\_is\_equal} has the following properties
+\begin{enumerate}
+    \item \textup{(Appropriate output domain).} If vector \texttt{v} of type \texttt{T} is in the input domain, then \texttt{function(v)} is in the output domain, otherwise the program will raise an \texttt{Exception}. 
+    \item \textup{(Stability Guarantee).} If two vector inputs $v, w$ are "$d_{in}$ - close" and if \texttt{Relation($d_{in}$, $d_{out}$) = True} then the corresponding outputs $function(v), function(w)$ are "$d_{out}$ - close". 
+\end{enumerate}
+\end{theorem}
+\begin{proof}
+\begin{enumerate}
+\item \textup{(Appropriate output domain).} Rust automatically enforces the type system to ensure it is in the correct output domain. The type signature of the function in the pseudo code and Rust code takes in a \texttt{Vec<T>} and returns \texttt{Vec<Bool>}. Since the rust code successfully compiles, by definition of the type signature the theorem must hold. Otherwise, the code will raise an exception for incorrect input type.
+
+
+
+\item \textup{(Stability Guarantee).}
+We can assume $d_{in} \leq d_{out}$ because Relation($d_{in}$, $d_{out}$) = \texttt{True}. We consider the symmetric distance as input metric. Since vector inputs $v, w$ are $d_{in}$-close, then by definition the symmetric distance of the multisets is bounded by $d_{in}$: $\abs{MultiSet(v) \Delta MultiSet(w)} \leq d_{in}$ 
```

Comment:
should put relation and MultiSet in the math mode but using fonts like mathrm, mathit, and/or texttt.  Otherwise LaTeX formats it as if it is a product of the variables R, e, l, a, t, i, o, n, and s.